// @jsdevtools/readdir-enhanced/lib/iterator/index.js
import { __exports as iterator } from '../../../../../_virtual/index3.js_commonjs-exports';
export { __exports as default } from '../../../../../_virtual/index3.js_commonjs-exports';
import require$$0 from 'fs';
import '../async/for-each.js';
import '../directory-reader.js';
import './pending.js';
import { __exports as forEach } from '../../../../../_virtual/for-each.js_commonjs-exports';
import { __exports as directoryReader } from '../../../../../_virtual/directory-reader.js_commonjs-exports';
import { __exports as pending } from '../../../../../_virtual/pending.js_commonjs-exports';

Object.defineProperty(iterator, "__esModule", { value: true });
iterator.readdirIterator = void 0;
const fs = require$$0;
const for_each_1 = forEach;
const directory_reader_1 = directoryReader;
const pending_1 = pending;
const iteratorFacade = { fs, forEach: for_each_1.asyncForEach };
function readdirIterator(dir, options) {
    let reader = new directory_reader_1.DirectoryReader(dir, options, iteratorFacade);
    let stream = reader.stream;
    let pendingValues = [];
    let pendingReads = [];
    let error;
    let readable = false;
    let done = false;
    stream.on("error", function streamError(err) {
        error = err;
        stream.pause();
        fulfillPendingReads();
    });
    stream.on("end", function streamEnd() {
        done = true;
        fulfillPendingReads();
    });
    stream.on("readable", function streamReadable() {
        readable = true;
        fulfillPendingReads();
    });
    return {
        [Symbol.asyncIterator]() {
            return this;
        },
        next() {
            let pendingRead = pending_1.pending();
            pendingReads.push(pendingRead);
            // eslint-disable-next-line @typescript-eslint/no-floating-promises
            Promise.resolve().then(fulfillPendingReads);
            return pendingRead.promise;
        }
    };
    function fulfillPendingReads() {
        if (error) {
            while (pendingReads.length > 0) {
                let pendingRead = pendingReads.shift();
                pendingRead.reject(error);
            }
        }
        else if (pendingReads.length > 0) {
            while (pendingReads.length > 0) {
                let pendingRead = pendingReads.shift();
                let value = getNextValue();
                if (value) {
                    pendingRead.resolve({ value });
                }
                else if (done) {
                    pendingRead.resolve({ done, value });
                }
                else {
                    pendingReads.unshift(pendingRead);
                    break;
                }
            }
        }
    }
    function getNextValue() {
        let value = pendingValues.shift();
        if (value) {
            return value;
        }
        else if (readable) {
            readable = false;
            while (true) {
                value = stream.read();
                if (value) {
                    pendingValues.push(value);
                }
                else {
                    break;
                }
            }
            return pendingValues.shift();
        }
    }
}
iterator.readdirIterator = readdirIterator;
